packages/api/src/submittable/createClass.ts

```diff
@@ -188,24 +189,37 @@ export function createClass <ApiType extends ApiTypes> ({ api, apiType, blockHas
     // send implementation for both immediate Hash and statusCb variants
     public send (statusCb?: Callback<ISubmittableResult>): SubmittableResultResult<ApiType> | SubmittableResultSubscription<ApiType> {
       const isSubscription = api.hasSubscriptions && (this.#ignoreStatusCb || !!statusCb);
+      const updatedInfo = extrinsicInfoMap.get(this);
 
       // eslint-disable-next-line @typescript-eslint/no-unsafe-return,@typescript-eslint/no-unsafe-call
       return decorateMethod(
         isSubscription
-          ? this.#observeSubscribe
-          : this.#observeSend
+          ? () => this.#observeSubscribe(updatedInfo)
+          : () => this.#observeSend(updatedInfo)
       )(statusCb);
     }
 
     /**
      * @description Signs a transaction, returning `this` to allow chaining. E.g.: `signAsync(...).send()`. Like `.signAndSend` this will retrieve the nonce and blockHash to send the tx with.
-     */
+    */
     public signAsync (account: AddressOrPair, partialOptions?: Partial<SignerOptions>): PromiseOrObs<ApiType, this> {
       // eslint-disable-next-line @typescript-eslint/no-unsafe-return,@typescript-eslint/no-unsafe-call
       return decorateMethod(
         (): Observable<this> =>
           this.#observeSign(account, partialOptions).pipe(
-            map(() => this)
+            map((info) => {
+              // If we got a full signed transaction from the signer, attach it
+              if (info.signedTransaction) {
+                const extrinsic = new Submittable(api.registry, info.signedTransaction);
```

Comment:
There is one problem: clients (e.g. pjs apps itself) might use `signAsync` + `send` in the form of 

``` 
const tx = ...
await tx.signAsync()
await tx.send()
```

So they disregard the returned value (which would be `extrinsic`) and submit the original one. Which would result in null value in `extrinsicInfoMap.get(this)`

Can we also do `extrinsicInfoMap.set(this, info);`?  This way both approaches would result in the same behavior and it would be easier to push clients to use this new feature by just asking them to update to the newest sdk version